src/lstm.js

```diff
@@ -97,10 +100,27 @@ export function lstm(input, weight, recurrentWeight, steps, hiddenSize,
     cellState = nextCell;
 
     if (returnSequence) {
-      nextHidden = reshape(nextHidden, [1, numDirections, null, hiddenSize]);
+      // Expand [numDirections, batchSize, hiddenSize] to
+      // [steps, numDirections, batchSize, hiddenSize]
+      nextHidden = reshape(nextHidden, [1, numDirections, batchSize, hiddenSize]);
+      // Concat output sequence along 0 axis (for steps dimension)
       sequence = (sequence ? concat([sequence, nextHidden], 0) : nextHidden);
     }
   }
 
+  if (direction === 'backward') {
+    // Reverse output sequence alog [0] axes (for steps dimension)
+    sequence = reverse(sequence, {axes: [0]});
```

Comment:
Please add a comment explaining why it needs to reverse. For example, refer to spec text, says "The sequence should contain every output from each time step in the temporal sequence, while the loop for steps concatenates sequence in a reversed order when direction is backward."

https://www.w3.org/TR/webnn/#api-mlgraphbuilder-lstm

Reply:
Tis convenient that the reverse operator exists such that you can just call it here.